### PR TITLE
Add a benchmark for GraphQL

### PR DESCRIPTION
**Pull request by kevingosse**
Note: this PR also changes the way the assembly is fetched in the integration

### CODE REVIEW

File: benchmarks/Benchmarks.Trace/GraphQLBenchmark.cs [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+        [Benchmark]
+        public int ExecuteAsync()
+        {
+            var task = (Task<ExecutionResult>)GraphQLIntegration.ExecuteAsync(Client, Context, (int)OpCodeValue.Callvirt, MdToken, (long)GuidPtr);
+
+            return task.GetAwaiter().GetResult().Value;
+        }
+
+        private class GraphQLClient
```
Comment (zacharycmontoya) [inline benchmarks/Benchmarks.Trace/GraphQLBenchmark.cs:56]: I'm guessing this class definition works for the benchmark, but can you also add that the class implements the interface that the method exists on?

```suggestion
        private class GraphQLClient : IExecutionStrategy
```

File: benchmarks/Benchmarks.Trace/GraphQLBenchmark.cs [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        {
+            public Task<ExecutionResult> ExecuteAsync(ExecutionContext context)
+            {
+                return Result;
+            }
+        }
+    }
+}
+
+namespace GraphQL
```
Comment (zacharycmontoya) [inline benchmarks/Benchmarks.Trace/GraphQLBenchmark.cs:66]: Do we need to redefine everything for the benchmark to work or can this be removed?

File: src/Datadog.Trace.ClrProfiler.Managed/Integrations/GraphQLIntegration.cs [hunk 3]
```diff
@@ -149,8 +149,7 @@ public static object ExecuteAsync(object executionStrategy, object context, int
 
             try
             {
-                var graphQLAssembly = AppDomain.CurrentDomain.GetAssemblies()
-                                               .Single(a => a.GetName().Name.Equals(GraphQLAssemblyName));
+                var graphQLAssembly = executionStrategy.GetType().Assembly;
```
Comment (zacharycmontoya) [inline src/Datadog.Trace.ClrProfiler.Managed/Integrations/GraphQLIntegration.cs:152]: I remember why this was implemented this way....the implementation type of this assembly may not exist in the `GraphQL` assembly. What we should really be doing here is getting the `Type` object that represents the interface `GraphQL.Execution.IExecutionStrategy` which definitely is in the `GraphQL` assembly.

```suggestion
                var executionStrategyInterfaceType = executionStrategy.GetInstrumentedInterface(GraphQLExecutionStrategyInterfaceName);
                var graphQLAssembly = executionStrategyInterfaceType.GetType().Assembly;
```

File: benchmarks/Benchmarks.Trace/GraphQLBenchmark.cs (hunk 2 above)
Comment (kevingosse) [reply benchmarks/Benchmarks.Trace/GraphQLBenchmark.cs:66]: Unfortunately this is needed

File: benchmarks/Benchmarks.Trace/GraphQLBenchmark.cs (hunk 1 above)
Comment (kevingosse) [reply benchmarks/Benchmarks.Trace/GraphQLBenchmark.cs:56]: Good catch!

File: src/Datadog.Trace.ClrProfiler.Managed/Integrations/GraphQLIntegration.cs (hunk 3 above)
Comment (kevingosse) [reply src/Datadog.Trace.ClrProfiler.Managed/Integrations/GraphQLIntegration.cs:152]: Done

File: benchmarks/Benchmarks.Trace/GraphQLBenchmark.cs [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+            new GraphQLBenchmark().ExecuteAsync();
+
+        }
+
+        [Benchmark]
+        public int ExecuteAsync()
+        {
+            var task = (Task<ExecutionResult>)GraphQLIntegration.ExecuteAsync(Client, Context, (int)OpCodeValue.Callvirt, MdToken, (long)GuidPtr);
+
+            return task.GetAwaiter().GetResult().Value;
```
Comment (tonyredondo) [inline benchmarks/Benchmarks.Trace/GraphQLBenchmark.cs:50]: just for curiosity, can't we add the Task<ExecutionResult> to the benchmark return type and await here?

Comment (kevingosse) [reply benchmarks/Benchmarks.Trace/GraphQLBenchmark.cs:50]: We probably can, but since I'm not 100% sure it's supported I feel safer that way 😅

**Review by tonyredondo — APPROVED** [review]

**Pull request merged**